internal/filters/main.go

```diff
@@ -0,0 +1,101 @@
+/*
+A CLI->Go template shim used to render arbitrary sprig functions.
+
+To Run:
+> go run main.go sprigFunctionName sprigFunctionArguments...
+
+Example:
+> go run main.go trim "      some string      "
+
+This program takes textual input and attempts to resolve the sprigFunctionArguments types through inspecting
+sprigFunctionName's signature.  Bad input results in fatal exit with error code 1.  Debug output is intentionally
+suppressed as stdout is utilized by the calling program.
+*/
+
+package main
+
+import (
+	"errors"
+	"fmt"
+	"github.com/Masterminds/sprig"
+	"os"
+	"reflect"
+	"strconv"
+)
+
+const fatalExitReturnCode = 1
+const funcArgsCLIArgsFirstIndex = 2
+const funcNameCLIArgsIndex = 1
+const funcReturnOutputIndex = 0
+
+// Wrapper to call a Golang function through reflection.
+func CallFunc(funcMap map[string]interface{}, name string, arguments ... interface{}) (result []reflect.Value,
+	err error) {
+
+	function := reflect.ValueOf(funcMap[name])
+	if len(arguments) != function.Type().NumIn() {
```

Comment:
NumIn panics if  the type's Kind is not Func. You should test .Type().NumIn()   and try to recover from panic and return gracefully and use golang recover() to catch the panic
func CallFunc()....{
defer func() {
        if e := recover(); e != nil {
            err = errors.New(name + " is not a callable fucntion.")
        }
    }()
   function := reflect.ValueOf(funcMap[name])
   function.Type().NumIn()

Reply:
I have an updated patch which does Kind checking to ensure that the **Value** is a **Func**.